Tests/FluentAssertions.Specs/Execution/AssertionScopeSpecs.cs

```diff
@@ -252,6 +252,129 @@ public void When_nested_scope_is_disposed_it_passes_reports_to_parent_scope()
+                    Id = Guid.NewGuid(),
+                    Children = new[]
+                    {
+                        new Container.Child1
+                        {
+                            Id = Guid.NewGuid(),
+                            Children = new[]
+                            {
+                                new Container.Child2
+                                {
+                                    StringProperty = "A",
+                                    IntProperty = 1,
+                                }
+                            }
+                        }
+                    }
+                }
+            };
+
+            var b = new[]
+            {
+                new Container
+                {
+                    Id = Guid.NewGuid(),
+                    Children = new[]
+                    {
+                        new Container.Child1
+                        {
+                            Id = Guid.NewGuid(),
+                            Children = new[]
+                            {
+                                new Container.Child2
+                                {
+                                    StringProperty = "B",
+                                    IntProperty = 1,
+                                }
+                            }
+                        }
+                    }
+                },
+                new Container
+                {
+                    Id = Guid.NewGuid(),
+                    Children = new[]
+                    {
+                        new Container.Child1
+                        {
+                            Id = Guid.NewGuid(),
+                            Children = new[]
+                            {
+                                new Container.Child2
+                                {
+                                    StringProperty = "B",
+                                    IntProperty = 1,
+                                }
+                            }
+                        }
+                    }
+                },
+            };
+
+            var messageWithoutMaxDepthRespected = "";
+
+            try
+            {
+                using (var scope = new AssertionScope())
+                {
+                    scope.FormattingOptions.MaxDepth = 1;
+                    a.Should().BeEquivalentTo(b);
+                }
+            }
+            catch (Exception ex)
+            {
+                messageWithoutMaxDepthRespected = ex.Message;
+            }
+
+            AssertionOptions.FormattingOptions.MaxDepth = 1;
+            var messageWithMaxDepthCorrectlyRespected = "";
+
+            try
+            {
+                a.Should().BeEquivalentTo(b);
+            }
+            catch (Exception ex)
+            {
+                messageWithMaxDepthCorrectlyRespected = ex.Message;
+            }
+            finally
+            {
+                AssertionOptions.FormattingOptions.MaxDepth = 5;
```

Comment:
In general this static property should not be writing to from within a test.
![image](https://github.com/fluentassertions/fluentassertions/assets/919634/8bcb05aa-0e67-4624-b438-ae84476c968d)

If we need to do so, the test need to be put in `When_global_formatting_settings_are_modified` which are run sequentially to avoid race conditions.

Reply:
Should this be a new class like `When_global_formatting_settings_are_modified `, or can I change this setup in the contructor to fit the needs of this test?

Reply:
Changed in [f657886](https://github.com/fluentassertions/fluentassertions/pull/2329/commits/f657886729f0a64dd26d96b26df9c2a79e811e55). Please have a look and check if this fits what you have desired, so I can squash this and the first commit afterwards.

Reply:
I'm sorry if I led you astray with my comment above, I hadn't thoroughly read through the issue and your fix before commenting.

An expansion of my "If we need to do so" is: we should be able to just [access the public API](https://blog.ploeh.dk/2015/09/22/unit-testing-internals/) and use simple [output verification](https://enterprisecraftsmanship.com/posts/styles-of-unit-testing/) to ensure we _test behavior rather than implementation_ (from BDD i suppose).
If we cannot write such a simple test and need to both access and mutate the global `AssertionOptions`, we need to ensure that it does not cause race-conditions. One such way could be to put it inside `When_global_formatting_settings_are_modified` whose tests are run sequentially and ensures to restore the previous state of `FormattingOptions` at the end of each test.

I seems it's our lucky day today as this test below fails without the fix and succeeds with the fix.

```cs
[Fact]
public void Formatting_options_flow_to_nested_assertion_scopes()
{
    // Arrange
    var a = new[]
    {
        new
        {
            Value = 42
        }
    };

    var b = new[]
    {
        new
        {
            Value = 42
        },
        new
        {
            Value = 42
        }
    };

    // Act
    using var scope = new AssertionScope();
    scope.FormattingOptions.MaxDepth = 1;
    a.Should().BeEquivalentTo(b);

    // Assert
    scope.Discard().Should().ContainSingle()
        .Which.Should().Contain("Maximum recursion depth of 1 was reached");
}
```

Reply:
Or we leave it like it is now, as this behaves the same 🤷‍♂️